aod-discord-bot.js

```diff
@@ -384,21 +384,24 @@ function getPermissionLevelForMember(member) {
 		perm = PERM_OWNER;
 	else {
 		//FIXME very inefficient
-		if (member.roles.cache.find(r => config.adminRoles.includes(r.name)))
-			perm = PERM_ADMIN;
-		else if (member.roles.cache.find(r => config.staffRoles.includes(r.name)))
-			perm = PERM_STAFF;
-		else if (member.roles.cache.find(r => config.divisionCommandRoles.includes(r.name)))
-			perm = PERM_DIVISION_COMMANDER;
-		else if (member.roles.cache.find(r => config.modRoles.includes(r.name)))
-			perm = PERM_MOD;
-		else if (member.roles.cache.find(r => r.name.endsWith('Officer')) ||
-			member.roles.cache.find(r => config.recruiterRoles.includes(r.name)))
-			perm = PERM_RECRUITER;
-		else if (member.roles.cache.find(r => r.name == config.memberRole))
-			perm = PERM_MEMBER;
-		else if (member.roles.cache.find(r => r.name == config.guestRole))
-			perm = PERM_GUEST;
+		const roleChecks = [
```

Comment:
Can we move the const to the global scope so it isn't rebuilt on every call?

Reply:
I take it back. Since you're accessing the config values, it does need to be recomputed every access (or at least recomputed when the config is updated).